krun/krun.c

```diff
@@ -0,0 +1,238 @@
+/*
+ * System calls for low latency MSR reads.
+ *
+ * Copyright (C) 2017, King's College London.
+ * Authored by Edd Barrett <vext01@gmail.com>.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include <linux/kernel.h>
+#include <linux/types.h>
+#include <linux/errno.h>
+#include <linux/smp.h>
+#include <asm/msr.h>
+#include <linux/uaccess.h>
+
+#include "krun_reg.h"
+
+/* protos */
+void krun_read_msrs(void *data);
+void krun_reset_msrs(void *ununsed);
```

Comment:
Did you mean `ununsed` to be called `unused`?

Reply:
I'd love to pretend that's kernel hacker terminology.. but...

Reply:
Does it mean something specific? I was baffled